### PR TITLE
Improve eventing istio resources lifecycle

### PR DESCRIPTION
**Pull request by pierDipi**
Improve handling of the transition enabled <-> disable for the istio feature flag.

### CODE REVIEW

File: openshift-knative-operator/pkg/eventing/extension.go [hunk 1]
```diff
@@ -44,10 +55,29 @@ func (e *extension) Manifests(ke base.KComponent) ([]mf.Manifest, error) {
...
 }
 
+func (e *extension) deleteResourcesSilently(m mf.Manifest) {
+	for _, np := range m.Resources() {
+		r /* plural */, _ /* singular  */ := meta.UnsafeGuessKindToResource(np.GroupVersionKind())
+		err := e.dynamicclient.Resource(r).
+			Namespace(np.GetNamespace()).
+			Delete(context.Background(), np.GetName(), metav1.DeleteOptions{})
+		if err != nil && !apierrors.IsNotFound(err) {
+			// Do not fail completely, just log the error
```
Comment (matzew) [inline openshift-knative-operator/pkg/eventing/extension.go:72]: lenient approach is good

**Review by matzew — APPROVED** [review]
/lgtm
/approve

**Comment by matzew** [comment]
Just curious:
did you ran into some issue/race w/ this transition of enabled/disabled, in the operator?

**Comment by pierDipi** [comment]
> Just curious: did you ran into some issue/race w/ this transition of enabled/disabled, in the operator?

No, I ran into the problem of having network policies that shouldn't be there as they blog traffic

**Pull request merged**